tt_metal/impl/dispatch/command_queue.cpp

```diff
@@ -356,10 +379,8 @@ const vector<DeviceCommand> EnqueueProgramCommand::assemble_runtime_args_command
             }
         }
 
-        uint32_t aligned_runtime_data_sizeB = align(runtime_args_len * sizeof(uint32_t), L1_ALIGNMENT) * num_packed_cmds;
-        uint32_t dispatch_cmd_sizeB = align(sizeof(CQDispatchCmd) + num_packed_cmds * sizeof(CQDispatchWritePackedMulticastSubCmd), L1_ALIGNMENT);
-        uint32_t mcast_payload_sizeB = dispatch_cmd_sizeB + aligned_runtime_data_sizeB;
-        uint32_t cmd_sequence_sizeB = align(sizeof(CQPrefetchCmd) +  mcast_payload_sizeB, PCIE_ALIGNMENT);
+        uint32_t mcast_payload_sizeB = get_runtime_payload_sizeB(num_packed_cmds, runtime_args_len, false);
```

Comment:
multicast of up to 255 common rt args is safe from this hazard, doesn't need same while-loop/treatment?

Reply:
good point, I think I misinterpreted the common rt args.I thought we wouldn't hit this case because common rt args would be using kernel groups. but we could have x kernels running on all cores with large numebr of common rt args. Ill add the same